src/mlpack/bindings/python/CMakeLists.txt

```diff
@@ -14,32 +14,46 @@ if (NOT BUILD_PYTHON_BINDINGS)
 endif ()
 
 # Generate Python setuptools file.
-find_package(PythonInterp)
-if (NOT PYTHON_EXECUTABLE)
-  not_found_return("Python not found; not building Python bindings.")
-else ()
-  message(STATUS "Found Python: ${PYTHON_EXECUTABLE}")
-endif ()
-
 # Import find_python_module.
 include(${CMAKE_SOURCE_DIR}/CMake/FindPythonModule.cmake)
+find_package(PythonInterp)
+if (NOT PYTHON_EXECUTABLE)
+  set(PY_NOT_FOUND_MSG "${PY_NOT_FOUND_MSG}\n    - Python")
+endif()
 find_python_module(distutils)
 if (NOT PY_DISTUTILS)
-  not_found_return("distutils not found; not building Python bindings.")
+  set(PY_NOT_FOUND_MSG "${PY_NOT_FOUND_MSG}\n    - distutils")
 endif ()
 find_python_module(Cython 0.24)
 if (NOT PY_CYTHON)
-  not_found_return("Cython not found; not building Python bindings.")
+  set(PY_NOT_FOUND_MSG "${PY_NOT_FOUND_MSG}\n    - Cython")
 endif ()
 find_python_module(numpy)
 if (NOT PY_NUMPY)
-  not_found_return("numpy not found; not building Python bindings.")
+  set(PY_NOT_FOUND_MSG "${PY_NOT_FOUND_MSG}\n    - numpy")
 endif ()
 find_python_module(pandas 0.15.0)
 if (NOT PY_PANDAS)
-  not_found_return("pandas not found; not building Python bindings.")
+  set(PY_NOT_FOUND_MSG "${PY_NOT_FOUND_MSG}\n    - pandas")
 endif ()
 
+## We need to check here if Python is even available.  Although actually
+## technically, I'm not sure if we even need to know!  For the tests though we
+## do.  So it's probably a good idea to check.
```

Comment:
I think this comment might be accurate for other languages (I think it was originally written for Julia or Go then adapted here?  I feel like I've seen it before), but not for Python.  We directly use setuptools in the build process, so it is necessary during the build process.  (The code is just fine, I'm only commenting on the comment. :))

Reply:
I am really sorry, I totally missed this. I have updated the comment in the last commit. :)